app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -84,8 +84,16 @@ def new_interface(attr = {})
       new_nic(attr)
     end
 
-    def new_volume(attr = { })
-      client.volumes.new(attr.merge(:allocation => '0G'))
+    def new_volume(attr = {})
+      if new_volume_errors.empty?
```

Comment:
Nitpick, use a guard clause like `return if new_volume_errors.empty?` to make it obvious what happens (the method returnsn il) and avoid having a method wrapped in an if condition